### PR TITLE
Dynamic Base Path

### PR DESCRIPTION
**Pull request by ItsMeBrianD**
### Checklist

- [ ] ~~For UI or styling changes, I have added a screenshot or gif showing before & after~~
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [ ] I have added to the docs where applicable
- [ ] ~~I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable~~

### CODE REVIEW

**Review by archiewood — COMMENTED** [review]
Can we add some docs to this?

Where does a user define basePath? `evidence.plugins.yaml`?

If so I'd suggest a section in https://docs.evidence.dev/deployment/overview/

File: .changeset/stale-kids-tan.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@evidence-dev/evidence': patch
+---
+
+Removed EVIDENCE_DISABLE_INCLUDE, behavior is now forced by `--debug`
```
Comment (zachstence) [inline .changeset/stale-kids-tan.md:5]: Would be better to leave these separate in my opinion. There are situations where we want `EVIDENCE_DISABLE_INCLUDE` without `--debug`, and vice versa.

File: packages/ui/core-components/src/lib/atoms/shadcn/badge/badge.svelte [hunk 2]
```diff
@@ -16,7 +16,6 @@
 
 <svelte:element
 	this={href ? 'a' : 'span'}
-	{href}
```
Comment (zachstence) [inline packages/ui/core-components/src/lib/atoms/shadcn/badge/badge.svelte:19]: Why is this change necessary?

File: packages/lib/sdk/src/utils/svelte/buildUrl.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    
+    let basePath = config.deployment.basePath;
+    if (basePath) {
+        if (!basePath?.startsWith('/')) {
+            basePath = `/${basePath}`;
+        }
+        if (basePath.endsWith('/')) {
+            basePath = basePath.slice(0, -1);
+        }
+        return `${basePath}${path}`;
```
Comment (zachstence) [inline packages/lib/sdk/src/utils/svelte/buildUrl.js:24]: This assumes that `path` always starts with `/`. Is this guaranteed to be true?

I think we should either
- Ensure append a `/` if `path` doesn't start with `/`, or
- Throw a descriptive error if `path` doesn't start with `/` rather than returning an incorrect URL (e.g. `/basetest` instead of `/base/test`)?

File: packages/lib/sdk/src/utils/svelte/buildUrl.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * Adjusts a path to include the configured base path
+ * Ignores undefined, and absolute URLs
+ * @param {string} path 
+ * @example buildUrl('http://localhost:3000/test') // 'http://localhost:3000/test'
+ * @example buildUrl('/test') // '/base/test'
+ * @example buildUrl(undefined) // undefined
+ * 
+ * @returns 
+ */
+export const buildUrl = (path) => {
```
Comment (zachstence) [inline packages/lib/sdk/src/utils/svelte/buildUrl.js:12]: I think this could be named better, maybe `addBasePath`? Something that describes what it does specifically

**Review by zachstence — CHANGES_REQUESTED** [review]

File: packages/ui/core-components/src/lib/atoms/shadcn/badge/badge.svelte (hunk 2 above)
Comment (ItsMeBrianD) [reply packages/ui/core-components/src/lib/atoms/shadcn/badge/badge.svelte:19]: This was in error

**Comment by zachstence** [comment]
Note for myself: Need to exclude base path from breadcrumbs
image

File: .changeset/stale-kids-tan.md (hunk 1 above)
Comment (zachstence) [reply .changeset/stale-kids-tan.md:5]: @ItsMeBrianD What do you think about this?

**Comment by hughess** [comment]
This looks great! All working on my end

**Review by hughess — APPROVED** [review]

File: .changeset/stale-kids-tan.md (hunk 1 above)
Comment (zachstence) [reply .changeset/stale-kids-tan.md]: Reverted this change: [0853ee0](https://github.com/evidence-dev/evidence/pull/2738/commits/0853ee02dbac3f3deac0d69a974be9afca59a953)

We can discuss more later if necessary

**Review by zachstence — APPROVED** [review]
